### ISSUE TITLE
Missing method Regex::captures_at

### ISSUE THREAD

**Issue by lilyball** [rust-lang/regex#547]
The `Regex` type has `_at` variants of most searching methods, except it doesn't have `captures_at`. This seems like an oversight.

**Comment by kballard** [comment]
I'm currently writing a Rust equivalent of Apple's [Scanner](https://developer.apple.com/documentation/foundation/scanner), because I want something that lets me quickly do ad-hoc parsing of input for Advent of Code, and one of the things I'm adding to it is a `scan_regex()` method. I wanted to be able to return a `regex::Captures` object, but also I thought it would be nice to allow the regex to know about surrounding context (since I'm matching it at an offset into the source string).

That said, since creating this issue, I realized this crate doesn't support look-around assertions, so the ability to see context pretty much just affects the behavior of `^` and `\A`, so ultimately it's not really very important.

But the core idea here is I'm taking a caller-supplied Regex and matching it in a substring of a caller-supplied larger string, and I want to return a `regex::Captures` to my caller, and it seems I can't do that right now while allowing the regex to see the surrounding context.

**Comment by BurntSushi** [comment]
Gotya. Yeah, that makes sense. A PR that adds it to `::Regex` and `::bytes::Regex` with a test or two would be welcome. :-) Otherwise, I can try to add it when I get around to it!

**Comment by rauschma** [comment]
This could work: https://docs.rs/fancy-regex/latest/fancy_regex/struct.Regex.html#method.captures_from_pos

**Comment by rauschma** [comment]
I have an additional requirement: I’d like to match at _exactly_ a given index. That could be achieved via either a flag (JavaScript has `y` for this purpose) or a special anchor that matches at the end of the previous match.

My use case is tokenization: https://users.rust-lang.org/t/matching-a-regex-at-a-particular-index/86070/2

**Comment by BurntSushi** [comment]
That is a separate issue entirely and is unlikely to be supported in the regex crate API. It will be available in regex-automata once #656 is done though, assuming I'm understanding you correctly. A concrete example with inputs, actual output and desired output would be much clearer.

**Comment by BurntSushi** [comment]
> a special anchor that matches at the end of the previous match.

I personally consider this a very nutty thing to do.

regex-automata will support this by simply making it possible to execute an anchored search anywhere. That is, `^` only matches at position `0`, but an anchored search is only required to produce matches starting at the place where the search started. They are orthogonal things, but it is what it sounds like you're after here.

**Comment by rauschma** [comment]
> A concrete example with inputs, actual output and desired output would be much clearer.

This is what I have so far (my first day of programming Rust, so it’s likely I have overlooked something; I’ll need the capture groups later):

```rs
use fancy_regex::Regex;
fn main() {
    let re = Regex::new(r"^((?P<emph>_)|(?P<ex>x))").unwrap();
    let text = "_x x";

    let mut index = 0usize;
    loop {
        match re.captures_from_pos(text, index).expect("TODO") {
            None => {break}
            Some(captures) => {
                let m = captures.get(0).unwrap();
                println!("TOKEN: {}", &text[m.start()..m.end()]);
                index = m.end();
            }
        }
    }
}
```

Actual output:

```
TOKEN: _
```

Output without `^`:

```
TOKEN: _
TOKEN: x
TOKEN: x
```

Desired output:

```
TOKEN: _
TOKEN: x
```
